tests/migrations/test_operations.py

```diff
@@ -624,9 +624,11 @@ def test_rename_model_with_self_referential_fk(self):
         self.assertIn(("test_rmwsrf", "horserider"), new_state.models)
         # Remember, RenameModel also repoints all incoming FKs and M2Ms
         self.assertEqual(
-            "test_rmwsrf.HorseRider",
+            'self',
             new_state.models["test_rmwsrf", "horserider"].fields[2][1].remote_field.model
```

Comment:
This has the nice side effect of avoiding transforming

```python
class Rider(models.Model):
    rider = models.ForeignKey('self')
```

Into
```python
class HorseRider(models.Model):
    rider = models.ForeignKey('app_label.HorseRider')
```

On a rename of `Rider` to `HorseRider`.